### DISCUSSION TITLE
[FEATURE REQUEST] Cross-platform file.mod_beacon

### DISCUSSION THREAD

**Discussion by BnGx** [saltstack/salt#67640] [Feature Requests]
Recent versions of Salt have added the possibility to introduce beacons using the `file.mod_beacon`, `pkg.mod_beacon`, `service.mod_beacon` functions.  Among these I seemed to notice that only `file.mod_beacon` is not cross platform so it is not possible to use this functionality in the various operating systems supported by Salt.

As there is already a watchdog beacon capable of exploiting this functionality, I propose to add the `beacon_type` option to the `file.mod_beacon` state where it is possible to specify `inotify` or `watchdog` to be able to take advantage of this opportunity in the cross platform environment.

Also I propose to use `watchdog` as the default value of `beacon_type`, or rather you could consider dynamically setting the default beacon to use in order to take advantages given by the specific architecture, for example `inotify` for inotify-capable systems and `watchdog` for other operating systems.

**Comment by OrangeDog** [comment]
>dynamically setting the default beacon to use 

That's the better idea
